pkg/gameserverallocations/controller.go

```diff
@@ -328,8 +342,95 @@ func (c *Controller) applyMultiClusterAllocation(gsa *v1alpha1.GameServerAllocat
 // allocateFromRemoteCluster allocates gameservers from a remote cluster by making
 // an http call to allocation service in that cluster.
 func (c *Controller) allocateFromRemoteCluster(gsa v1alpha1.GameServerAllocation, connectionInfo *multiclusterv1alpha1.ClusterConnectionInfo, namespace string) (*v1alpha1.GameServerAllocation, error) {
-	// TODO: implement getting secrets and making rest call to remote cluster
-	return nil, nil
+	var gsaResult v1alpha1.GameServerAllocation
+
+	// TODO: handle converting error to apiserver error
+	client, err := c.createRemoteClusterRestClient(namespace, connectionInfo.SecretName)
+	if err != nil {
+		return nil, err
+	}
+
+	// Forward the game server allocation request to another cluster,
+	// and disable multicluster settings to avoid the target cluster
+	// forward the allocation request again.
+	gsa.Spec.MultiClusterSetting.Enabled = false
+	body, err := json.Marshal(gsa)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: Retry on transient error --> response.StatusCode >= 500
+	response, err := client.Post(connectionInfo.AllocationEndpoint, "application/json", bytes.NewBuffer(body))
+	if err != nil {
+		return nil, err
+	}
+	defer response.Body.Close() // nolint: errcheck
+
+	data, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, err
+	}
+	if response.StatusCode >= 400 {
+		return nil, errors.New(string(data))
+	}
+
+	err = json.Unmarshal(data, &gsaResult)
```

Comment:
nit:

this can help avoid having to explicitly read `data`:

```
if err := json.NewUnmarshaler(response.Body).Decode(&gsaResult); err != nil {
  return nil, fmt.Errors("unable to parse request: %v", err)
}
```

Reply:
The problem is this would fail for error cases that does not decode to gsa. So it actually need the two step deserialization.

Reply:
not sure what's going on here, then

is it important to return data in error string? if so, can you add some comments?

Reply:
The data is returned as an error string only for http status 4xx and 5xx. In that case the response body is expected to contain error and not GSA message. Therefore Decode(&gsaResult) would fail to parse the response body as GSA.
I'll add comment in the next PR since this is already merged.